### PR TITLE
Update source code for external and modules

### PR DESCRIPTION
**Pull request by yong171966**
Change-Id: I80bac554c6ffea7a49e464a73a149cdb0e8cd61b

### CODE REVIEW

**Comment by skylarCai** [comment]
感谢提交 Pull Requests！
Thanks for submitting a pull request.

**Comment by skylarCai** [comment]
auto build fail,number:215

**Review by yanxiaoyong-ch — COMMENTED** [review]
this looks ok for me

File: external/activation/component.mk [hunk 1]
```diff
@@ -7,5 +7,8 @@ COMPONENT_SUBMODULES += $(COMPONENT_DIR)
 COMPONENT_ADD_INCLUDEDIRS := include
 COMPONENT_SRCDIRS := src
 
-SRCDIRS_HAAS += $(addprefix $(EXTERNAL_DIR)/$(COMPONENT_DIR)/, $(COMPONENT_SRCDIRS))
-INCLUDEDIRS_HAAS += $(addprefix $(EXTERNAL_DIR)/$(COMPONENT_DIR)/, $(COMPONENT_ADD_INCLUDEDIRS))
\ No newline at end of file
+INC_HAAS += $(addprefix $(EXTERNAL_DIR)/$(COMPONENT_DIR)/, $(COMPONENT_ADD_INCLUDEDIRS))
```
Comment (skylarCai) [inline external/activation/component.mk:10]: LGFM

**Review by skylarCai — COMMENTED** [review]
NO SECURITY RISK

**Pull request merged**